ci/docker_push.sh

```diff
@@ -20,6 +20,12 @@ then
    docker push lyft/envoy:latest
    docker tag lyft/envoy:latest lyft/envoy:$CIRCLE_SHA1
    docker push lyft/envoy:$CIRCLE_SHA1
+   if [ -z "$CIRCLE_TAG" ]
```

Comment:
I think you want `-n` (test that the string is non-zero length) as opposed to `-z` (tests for zero length string).